src/api/NodeResolver.ts

```diff
@@ -21,9 +23,28 @@ export class NodeResolver {
     async nodes(
         @Arg("ids", () => [String], { nullable: true }) ids?: string[],
         @Arg("pageSize", () => Int, { nullable: true }) pageSize?: number,
-        @Arg("cursor", { nullable: true }) cursor?: string,
+        @Arg("cursor", { nullable: true }) cursor?: string
     ): Promise<DeepOmit<Nodes, { items: { location: never }[] }>> {
-        return this.repository.getNodes(ids, pageSize, cursor)
+        return this.repository.getNodes(
+            (ids !== undefined) ? ids.map((id) => id as DhtAddress) : undefined,
```

Comment:
Unnecessary map invocation?
```suggestion
            (ids !== undefined) ? ids as DhtAddress[] : undefined,
```